packages/coreutils/src/debounce.ts

```diff
@@ -0,0 +1,24 @@
+import { PromiseDelegate } from '@phosphor/coreutils';
+
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+/**
+ * Returns a debounced function that can be called multiple times safely and
+ * only executes the underlying function once per `interval`.
+ *
+ * @param fn - The function to debounce.
+ *
+ * @param interval - The debounce interval; defaults to 500ms.
+ */
+export function debounce(fn: () => any, interval = 500): () => Promise<void> {
```

Comment:
Maybe this function should be templated to resolve with the return type of `fn`?

Reply:
👍